core/src/main/scala/org/http4s/Header.scala

```diff
@@ -132,7 +132,16 @@ object Header {
     _.toString
   }
 
-  implicit val HeaderEq: Eq[Header] = Eq.instance[Header] { (a, b) =>
-    a.name === b.name && a.value === b.value
-  }
+  @deprecated(message = "Please use HeaderOrder instead", since = "1.0.0-M8")
```

Comment:
We can do this in series/0.21.  I can cherry-pick it if you don't want to rebase it, but we should change the version number.

Reply:
@rossabaker I'll rebase.